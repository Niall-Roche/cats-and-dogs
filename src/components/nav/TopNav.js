import {styled} from 'styled-components'
import {CAT_DOG_EMOJI} from '../../constants/images'
import {Link} from 'react-router-dom'

const TopNavContainer = styled.div`
    position: fixed;
    background-color: ${props => props?.theme?.accent};
    width: 100%;
    align-items: center;
    height: 50px;
    z-index: 1;
    /* padding-top: 10px; */
    /* padding-bottom: 10px; */
`

const StyledLink = styled(Link)`
    text-decoration: none;
    color: ${props => props?.theme?.primary};
`

const BrandImage = styled.img`
    margin-left: 10px;
    margin-right: 10px;
`

const Title = styled.span`
    font-weight: bold;
`

const SubTitle = styled.span`
    margin-right: 10px;
`

const TopNav = () => {
  return (
    <TopNavContainer className='d-flex flex-row mb-1'>
      <StyledLink className='d-flex align-items-center' to='/'>
        <BrandImage src={CAT_DOG_EMOJI} width={32} height={32} />
        <Title>Cats & Dogs</Title>
      </StyledLink>
      <SubTitle className='ms-auto'>Powered By GIPHY</SubTitle>
    </TopNavContainer>
  )
}

export default TopNav