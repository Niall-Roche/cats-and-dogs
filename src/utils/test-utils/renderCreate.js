import {ThemeProvider} from '@tanstack/react-query-devtools/build/lib/theme'
import TestRenderer from 'react-test-renderer'
import theme from '../../app-theme.json'
import {CatsAndDogsProvider} from '../../hooks/useCatsAndDogs'
import {PaginationProvider} from '../../hooks/usePagination'

export default comp => TestRenderer.create(
  <CatsAndDogsProvider>
    <PaginationProvider>
      <ThemeProvider theme={theme}>
        {comp}
      </ThemeProvider>
    </PaginationProvider>
  </CatsAndDogsProvider>
)