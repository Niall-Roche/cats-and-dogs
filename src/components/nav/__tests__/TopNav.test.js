import React from 'react'
import TopNav from '../TopNav'
import 'jest-styled-components'
import renderCreate from '../../../utils/test-utils/renderCreate'
import {BrowserRouter} from 'react-router-dom'

describe('TopNav', () => {
  it('should match snapshot', () => {
    const comp = renderCreate(
      <BrowserRouter>
        <TopNav />
      </BrowserRouter>
    )
    const tree = comp.toJSON()

    expect(tree).toMatchSnapshot()
  })
})