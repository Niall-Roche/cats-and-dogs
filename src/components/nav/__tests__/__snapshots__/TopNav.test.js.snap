// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav should match snapshot 1`] = `
<div
  className="jcNbAj d-flex flex-row mb-1"
>
  <a
    className="dSLzlO d-flex align-items-center"
    href="/"
    onClick={[Function]}
  >
    <img
      className="fDLtHd"
      height={32}
      src="catdog.png"
      width={32}
    />
    <span
      className="eHVcVy"
    >
      Cats & Dogs
    </span>
  </a>
  <span
    className="dNffVX ms-auto"
  >
    Powered By GIPHY
  </span>
</div>
`;
